core/src/avm2/globals/array.rs

```diff
@@ -334,7 +333,7 @@ pub fn to_locale_string<'gc>(
 /// array while this happens would cause a panic; this code exists to prevent
 /// that.
 pub struct ArrayIter<'gc> {
-    array_object: Object<'gc>,
+    array_object: Value<'gc>,
```

Comment:
Still think this should be an Object, especially considering its main use point might be converted to `get_index_property` eventually.